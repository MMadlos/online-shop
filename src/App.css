/* VARIABLES */
:root {
	/* COLORS */
	--black: #333333;
	--white: #fafafa;
	--accent-purple: #7443ff;
	--light-purple: #e3d9ff;
	--accent-yellow: #e2fe91;
	--yellow-star: #ffda17;
	--light-grey: #f2f2f2;
	--grey: #e5e6ea;
	--dark-grey: #c7c9d1;
	--light-black: #72778d;
	--alert: #fa7373;
	--light-green: #e3fee4;
	--green: #15a409;

	/* OTHERS */
	--padding-section-laterals: 15vw;
	--padding-media-small: 16px;
	--font-main: "Inter", sans-serif;
	--radial-topleft-purple: radial-gradient(circle at top left, var(--light-purple) 25%, var(--white) 100%);
	--radial-topright-purple: radial-gradient(circle at top right, var(--light-purple) 25%, var(--white) 100%);
}

/* GENERIC */

body {
	font-family: var(--font-main);
}

#root {
	/* Added to grow the main content when it doesn't fill the available heigh*/
	height: 100vh;

	display: flex;
	flex-direction: column;
}

footer {
	background-color: var(--black);
	color: var(--white);
	padding: 20px 10vw;
}

h1 {
	font-size: 1.5rem;
	font-weight: 900;
}

h2 {
	font-size: 1.25rem;
	font-weight: 800;
}

ul {
	list-style-type: none;
}

a {
	text-decoration-line: none;
	color: var(--black);
}

button {
	outline: none;
	border: none;
}

@media (max-width: 500px) {
	h1 {
		font-size: 16px;
	}
}

/* ACTION BAR */
.action-bar {
	display: flex;
	justify-content: space-between;

	background-color: var(--white);
	padding: 16px var(--padding-section-laterals);
	padding-bottom: 0px;

	> form {
		display: flex;
		align-items: center;
		gap: 16px;

		padding: 4px 16px;
		background-color: var(--white);

		border: 1px solid var(--dark-grey);
		border-radius: 4px;

		width: 80%;
		max-width: 600px;

		> i {
			color: var(--dark-grey);
		}

		> input {
			border: none;
			background-color: var(--white);
			color: var(--black);
			font-size: 14px;
			width: 100%;

			&::placeholder {
				color: var(--dark-grey);
			}

			&:focus {
				outline: none;
			}

			&::-webkit-search-cancel-button {
				cursor: pointer;
			}
		}
	}

	#search-not-found {
		color: red;
	}
}

@media (max-width: 500px) {
	.action-bar {
		flex-direction: column;
		gap: 8px;
		padding: 12px var(--padding-media-small);

		> form {
			gap: 16px;
			padding: 8px;
			width: 100%;
		}
	}
}

.dropdown-container {
	position: relative;

	> button#sort {
		display: flex;
		justify-content: space-between;
		align-items: center;
		gap: 32px;
		padding: 8px 16px;

		background-color: var(--grey);
		border: 1px solid var(--dark-grey);
		border-radius: 2px;

		color: var(--light-black);
		font-weight: 600;

		&:hover,
		&.selected {
			background-color: var(--light-black);
			color: var(--white);
		}

		> i {
			font-size: 1rem;
		}
	}

	.dropdown-menu {
		position: absolute;
		transform-origin: top right;
		right: 0;
		width: 250px;

		background-color: var(--white);

		display: flex;
		flex-direction: column;
		justify-content: center;
		align-items: start;

		border: 1px solid var(--dark-grey);
		border-radius: 2px;

		padding: 8px;
		margin-top: 8px;

		.item {
			color: var(--black);
			width: 100%;
			padding: 8px 16px;
			text-align: left;
			border-radius: 2px;
			background-color: transparent;

			&:hover {
				background-color: var(--light-grey);
			}
		}

		.selected {
			background-color: var(--grey);
			color: var(--black);
			font-weight: 600;
		}

		.group-title {
			font-size: 12px;
			font-weight: 700;
			color: var(--light-black);
		}

		.divider {
			height: 1px;
			background-color: var(--grey);
			width: 100%;
			margin: 8px 0;
		}
	}
}

@media (max-width: 500px) {
	.dropdown-container {
		> button#sort {
			gap: 8px;
			padding: 8px;
			width: 100%;
			font-weight: 500;

			> i {
				font-size: 0.75rem;
			}
		}

		.dropdown-menu {
			transform-origin: top left;
			left: 0;
			width: 100%;
		}
	}
}

.action-messages {
	display: flex;
	justify-content: space-between;
	padding: 0 var(--padding-section-laterals);
	height: 50px;
	background-color: var(--white);

	.search-not-found {
		color: var(--alert);
		font-size: 14px;
		font-weight: 500;
		width: 100%;
	}

	.chip-sort {
		color: var(--black);
		font-size: 12px;
		width: 100%;
		text-align: right;

		display: flex;
		justify-content: end;
		align-items: center;
		gap: 12px;
		padding-top: 16px;

		.sort-selected {
			display: flex;
			justify-content: space-between;
			align-items: center;
			gap: 12px;

			background-color: var(--grey);
			padding: 4px 16px;
			border: 1px solid var(--dark-grey);
			border-radius: 50px;

			> i {
				color: var(--light-black);

				&:hover {
					color: var(--dark-grey);
				}
			}
		}
	}
}

@media (max-width: 500px) {
	.action-messages {
		flex-direction: column;
		padding: 0 var(--padding-media-small);
		gap: 8px;
		height: fit-content;

		.search-not-found {
			font-size: 12px;
			font-weight: 700;
		}

		.chip-sort {
			font-size: 12px;
			width: 100%;

			justify-content: flex-start;
			padding-top: 0;
			margin-bottom: 8px;

			.sort-selected {
				gap: 12px;

				padding: 4px 12px;

				> i {
					font-size: 1rem;
					color: var(--light-black);
				}
			}
		}
	}
}

/* PRODUCT INFO */
section.product-info {
	padding: 16px var(--padding-section-laterals);
	color: var(--black);

	> a {
		display: flex;
		align-items: center;
		gap: 8px;

		font-size: 14px;
		font-weight: 500;
		color: var(--light-black);

		&:hover {
			color: var(--black);
		}
	}

	.product-details-container {
		padding-top: 40px;
		display: flex;
		gap: 80px;

		.image-container {
			min-width: 40%;
			max-height: 65vh;
			max-width: 60%;

			> img {
				object-fit: contain;
				object-position: center;
				height: 100%;
				width: 100%;
			}
		}

		.product-details {
			display: flex;
			flex-direction: column;
			justify-content: space-between;

			background-color: var(--light-grey);
			padding: 40px;
			border-radius: 8px;
			min-width: 300px;
			min-height: 50vh;

			.details-container {
				display: flex;
				flex-direction: column;
				gap: 16px;

				> h1 {
					font-size: 2rem;
					font-weight: 800;
				}

				.rate-container {
					display: flex;
					align-items: center;
					gap: 8px;
					color: var(--light-black);
					font-size: 14px;
					font-weight: 600;
					> i {
						color: var(--yellow-star);
					}
				}

				.description {
					font-size: 14px;
					line-height: 150%;
					font-weight: 500;
					letter-spacing: 0.25px;
					color: var(--light-black);
				}

				.price {
					font-size: 24px;
					font-weight: 700;
					color: var(--light-black);
				}
			}

			.add-cart-container {
				display: flex;
				justify-content: space-between;
				align-items: center;
				flex-wrap: wrap;
				gap: 40px;
				row-gap: 16px;

				.counter {
					min-width: fit-content;
					max-width: 100%;

					display: flex;
					align-items: center;
					gap: 8px;

					#increase,
					#decrease {
						font-size: 14px;
						color: var(--black);
						background-color: var(--grey);
						padding: 8px;
						width: 35px;
						aspect-ratio: 1 / 1;

						border-radius: 4px;

						&:hover:not(:disabled) {
							color: var(--black);
							background-color: var(--dark-grey);
						}

						&:disabled {
							opacity: 50%;

							> i {
								opacity: 25%;
							}
						}
					}

					.quantity {
						display: flex;
						align-items: center;
						justify-content: center;
						padding: 8px 24px;
						font-weight: 600;
						background-color: var(--white);
						border-radius: 4px;
						width: 100%;
					}
				}

				> button#add-cart {
					min-width: fit-content;
					max-width: 100%;
					flex-grow: 1;
					font-weight: 700;
					font-size: 14px;
					padding: 12px 24px;

					border-radius: 2px;
					color: var(--white);
					background-color: var(--black);

					&:hover {
						cursor: pointer;
						background-color: var(--light-black);
					}
				}
			}
		}
	}
}

@media (max-width: 500px) {
	section.product-info {
		padding: 16px var(--padding-media-small);
		color: var(--black);

		.product-details-container {
			padding-top: 16px;
			flex-direction: column;
			align-items: center;
			gap: 24px;

			.image-container {
				height: 150px;

				> img {
					object-fit: contain;
					object-position: center;
				}
			}

			.product-details {
				padding: var(--padding-media-small);
				gap: 16px;
				min-height: fit-content;

				.details-container {
					display: flex;
					flex-direction: column;
					gap: 12px;

					> h1 {
						font-size: 1.25rem;
						font-weight: 700;
					}

					.rate-container {
						font-size: 12px;
						font-weight: 500;
					}

					.description {
						line-height: 125%;
						font-weight: 500;
						letter-spacing: 0.15px;
					}

					.price {
						font-size: 1rem;
					}
				}

				.add-cart-container {
					.counter {
						min-width: fit-content;
						width: 100%;

						display: flex;
						align-items: center;
						gap: 8px;

						#increase,
						#decrease {
							font-size: 12px;
							padding: 8px 24px;
							/* width: 35px; */
							aspect-ratio: 1/1;

							border-radius: 50px;

							> i {
								display: flex;
								justify-content: center;
								color: var(--light-black);
							}
						}

						.quantity {
							display: flex;
							align-items: center;
							justify-content: center;
							padding: 8px 24px;
							font-weight: 600;
							border-radius: 4px;
							width: 100%;
						}
					}
				}

				/* .quantity {
					font-size: 14px;
				} */
			}
		}
	}
}

/* SECTION - CART */
section#cart {
	padding: 16px var(--padding-section-laterals) 40px var(--padding-section-laterals);

	display: flex;
	flex-direction: column;
	gap: 8px;
	color: var(--black);

	.return-container {
		display: flex;
		gap: 8px;
		padding: 8px 8px 8px 0;
		align-items: center;
		font-weight: 500;

		&:hover {
			cursor: pointer;
			opacity: 50%;
		}
	}

	> h1 {
		margin-top: 24px;
		font-weight: 700;
	}

	.product-card.cart {
		background-color: var(--white);
		border: 1px solid var(--dark-grey);
		border-radius: 8px;
		padding: 8px;

		&:hover {
			border-color: var(--light-black);
		}

		.product-info {
			display: flex;
			justify-content: space-between;
			align-items: center;
			gap: 32px;

			padding: 8px 16px;

			.product-info-details-container {
				display: flex;
				gap: 24px;
				max-width: 60%;

				> img {
					--side: 100px;
					max-height: var(--side);
					max-width: var(--side);
					aspect-ratio: 1 / 1;

					object-position: center;
					object-fit: contain;
				}

				.produt-info-name-price-container {
					display: flex;
					flex-direction: column;
					gap: 8px;

					.name {
						font-size: 16px;
						font-weight: 700;
					}

					.description {
						font-size: 14px;
						color: var(--light-black);

						-webkit-line-clamp: 2;
						-webkit-box-orient: vertical;
						overflow: hidden;
						text-overflow: ellipsis;
						display: -webkit-box;
					}

					.price {
						font-size: 1.25rem;
						font-weight: 600;
					}
				}
			}

			.counter {
				min-width: fit-content;
				max-width: 100%;

				display: flex;
				align-items: center;
				gap: 8px;

				#increase,
				#decrease {
					font-size: 12px;
					color: var(--black);
					background-color: var(--grey);
					padding: 8px;
					width: 35px;
					aspect-ratio: 1 / 1;

					border-radius: 4px;

					&:hover:not(:disabled) {
						color: var(--black);
						background-color: var(--dark-grey);
					}

					&:disabled {
						opacity: 50%;

						> i {
							opacity: 25%;
						}
					}
				}

				.quantity {
					display: flex;
					align-items: center;
					justify-content: center;
					padding: 8px 24px;
					font-weight: 600;
					border-radius: 4px;
					width: 100%;
				}
			}

			#remove-cart {
				font-weight: 700;
				font-size: 14px;
				padding: 8px 24px;
				width: fit-content;
				border-radius: 2px;

				color: var(--light-black);
				background-color: var(--grey);
				font-weight: 500;

				&:hover {
					cursor: pointer;
					background-color: var(--dark-grey);
				}
			}
		}
	}

	.check-out-container {
		display: flex;
		justify-content: space-between;
		align-items: center;

		gap: 16px;
		padding: 16px 16px;

		border-radius: 6px;
		background-color: var(--light-grey);

		.total-container > p {
			font-size: 1.25rem;

			> span {
				padding-left: 16px;
				font-weight: 700;
			}
		}

		> button#checkout {
			font-size: 16px;
			font-weight: 700;
			padding: 8px 32px;
			border-radius: 2px;
			letter-spacing: 0.25px;

			color: var(--white);
			background-color: var(--accent-purple);

			&:hover {
				cursor: pointer;
				opacity: 50%;
			}
		}
	}

	.cart-emtpy {
		display: flex;
		justify-content: center;
		padding: 40px 0;
		border: 4px;
		color: var(--light-black);
		background-color: var(--light-grey);
	}
}

@media (max-width: 500px) {
	section#cart {
		padding: 16px var(--padding-media-small);

		display: flex;
		flex-direction: column;
		gap: 8px;
		color: var(--black);

		.return-container {
			padding: 0;
			font-size: 14px;
			color: var(--light-black);

			> a {
				font-weight: 500;
				color: var(--light-black);
			}
		}

		> h1 {
			margin-top: 8px;
			font-weight: 700;
		}

		> a {
			.product-card.cart {
				.product-info {
					/* display: flex; */
					flex-direction: column;
					/* justify-content: space-between; */
					/* align-items: center; */
					gap: 16px;

					/* padding: 8px 16px; */

					.product-info-details-container {
						/* display: flex; */
						gap: 24px;
						max-width: 100%;

						flex-direction: column;

						> img {
							max-height: 20vh;
							max-width: 100%;

							object-position: center;
							object-fit: contain;
						}

						.produt-info-name-price-container {
							display: flex;
							flex-direction: column;
							gap: 8px;

							.name {
								font-size: 16px;
								font-weight: 700;

								-webkit-line-clamp: 2;
								-webkit-box-orient: vertical;
								overflow: hidden;
								text-overflow: ellipsis;
								display: -webkit-box;
							}

							.description {
								display: none;
							}

							.price {
								font-size: 1rem;
								font-weight: 500;
								color: var(--light-black);
							}
						}
					}

					.counter {
						min-width: fit-content;
						width: 100%;

						display: flex;
						align-items: center;
						gap: 8px;

						#increase,
						#decrease {
							font-size: 12px;
							padding: 8px 24px;
							/* width: 35px; */
							aspect-ratio: 1/1;

							border-radius: 50px;

							> i {
								display: flex;
								justify-content: center;
								color: var(--light-black);
							}
						}

						.quantity {
							display: flex;
							align-items: center;
							justify-content: center;
							padding: 8px 24px;
							font-weight: 600;
							border-radius: 4px;
							width: 100%;
						}
					}

					#remove-cart {
						width: 100%;
						border-radius: 4px;
					}
				}
			}
		}

		.check-out-container {
			flex-direction: column;
			padding: 16px 8px;

			> button#checkout {
				width: 100%;
			}
		}
	}
}

/* RECOMMENDATION SECTION */

section.recommendation {
	--padding-top-bottom: 40px;

	padding: var(--padding-top-bottom) var(--padding-section-laterals);
	background-color: var(--grey);

	> h2 {
		font-size: 1.5rem;
		font-weight: 600;
		color: var(--accent-purple);
		margin-bottom: 40px;
	}

	.recommendation-container {
		display: grid;
		grid-template-columns: repeat(4, 1fr);
		align-items: center;
		gap: 16px;

		> a {
			height: 100%;

			.product-card {
				background-color: var(--light-grey);
				width: 100%;

				overflow: hidden;
				display: flex;
				flex-direction: column;
				height: 100%;

				border: 1px solid var(--dark-grey);
				border-radius: 4px;

				&:hover {
					border: 1px solid var(--black);
				}

				> img {
					min-height: 150px;
					max-height: 250px;
					height: 100%;
					object-fit: contain;
					object-position: center;
					padding: 32px 64px;
					background-color: var(--white);
				}

				.product-info {
					display: flex;
					justify-content: space-between;

					flex-direction: column;
					gap: 16px;
					background-color: var(--light-grey);
					padding: 16px;

					.product-info-container {
						display: flex;
						flex-direction: column;
						gap: 12px;

						#name {
							-webkit-line-clamp: 2;
							-webkit-box-orient: vertical;
							overflow: hidden;
							text-overflow: ellipsis;
							display: -webkit-box;
							font-size: 1rem;
							font-weight: 600;
							line-height: 130%;
						}

						.rate-container {
							display: flex;
							align-items: center;
							margin-top: -8px;
							gap: 8px;
							font-size: 12px;
							color: var(--light-black);

							.fa-star {
								color: var(--yellow-star);
							}
						}

						#price {
							font-size: 14px;
							font-weight: 600;
							color: var(--light-black);
						}
					}
				}
			}
		}
	}
}

@media (max-width: 500px) {
	section.recommendation {
		padding: var(--padding-top-bottom) var(--padding-media-small);
		background-color: var(--white);

		> h2 {
			font-size: 1.25rem;
			font-weight: 600;
			margin-bottom: 16px;
		}

		.recommendation-container {
			display: grid;
			grid-template-columns: repeat(4, 1fr);
			align-items: center;
			gap: 8px;

			overflow-x: scroll;

			> a {
				height: 100%;

				.product-card {
					background-color: var(--light-grey);
					min-width: 200px;

					border-radius: 8px;

					&:hover {
						border: 1px solid var(--black);
					}

					> img {
						height: 200px;
						width: 100%;

						padding: 16px;
					}
				}
			}
		}
	}
}

/* .home {
	background-color: var(--accent-yellow);

	> main {
		background-color: transparent;
	}
} */
