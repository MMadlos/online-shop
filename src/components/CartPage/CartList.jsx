import { Fragment } from "react"
import CartProduct from "./CartProduct"

function CartList({ cartList, onClickEdit }) {
	return (
		<div className="cart-list-container">
			{cartList?.map((product, index) => {
				const { id } = product
				const isLastProduct = index === cartList.length - 1

				return (
					<Fragment key={id}>
						<CartProduct
							product={product}
							onClickEdit={onClickEdit}
						/>
						{!isLastProduct && <div className="divider"></div>}
					</Fragment>
				)
			})}
		</div>
	)
}

export default CartList
