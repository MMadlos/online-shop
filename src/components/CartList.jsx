import { Link } from "react-router-dom"
import { useContext } from "react"
import { ShopContext } from "../App"

import Counter from "./Atoms/Counter"

function CartProduct({ product }) {
	const { handleClickRemoveCart } = useContext(ShopContext)

	const { id, name, price, url, quantity, description } = product

	return (
		<Link to={`/product/${id}`}>
			<div
				key={id}
				className="product-card cart"
				data-product-id={id}>
				<div className="product-info">
					<div className="product-info-details-container">
						<img
							src={url}
							alt=""
						/>

						<div className="produt-info-name-price-container">
							<p className="name">
								<strong>{name}</strong>
							</p>
							<p className="description">{description}</p>
							<p className="price">$ {price}</p>
						</div>
					</div>

					<Counter
						context="cart"
						currentQuantity={quantity}
					/>
					<button
						id="remove-cart"
						onClick={handleClickRemoveCart}>
						Remove
					</button>
				</div>
			</div>
		</Link>
	)
}

function CartList() {
	const { cartList } = useContext(ShopContext)

	const sumCart = cartList.reduce((acc, curr) => acc + curr.price * curr.quantity, 0)
	const totalToPay = sumCart.toFixed(2)

	return (
		<section id="cart">
			<div className="return-container">
				<i className="fa-solid fa-chevron-left"></i>
				<Link to="/">Return to shop</Link>
			</div>

			<h1>Order</h1>

			{cartList.length === 0 && <h2> THERE IS NO PRODUCT IN THE CART</h2>}

			{cartList?.map((product) => {
				return (
					<CartProduct
						key={product.id}
						product={product}
					/>
				)
			})}

			<div className="check-out-container">
				<div className="total-container">
					<p>
						Total amount:
						<span>$ {totalToPay}</span>
					</p>
				</div>
				<button id="checkout">Checkout</button>
			</div>
		</section>
	)
}

export default CartList
