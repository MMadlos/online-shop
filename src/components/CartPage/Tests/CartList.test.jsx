import { describe, expect, it, vi } from "vitest"
import { render, screen } from "@testing-library/react"
import { BrowserRouter } from "react-router-dom"

import { ShopContext } from "../../../App"
import { testMockMultipleProducts } from "../../../consts/testMockProduct"

import CartList from "../CartList"

vi.mock("../CartProduct", () => ({
	default: ({ product, onClickEdit }) => (
		<div>
			<div data-testid="product-id">{product.id}</div>
			<div data-testid="product-name">{product.name}</div>
			<button
				data-testid="button"
				onClick={onClickEdit}>
				Click
			</button>
		</div>
	),
}))

describe("CartList", () => {
	it.skip("renders the component", () => {
		const cartList = { cartList: testMockMultipleProducts }

		const container = render(
			<ShopContext.Provider value={cartList}>
				<BrowserRouter>
					<CartList />
				</BrowserRouter>
			</ShopContext.Provider>
		)

		expect(container).toMatchSnapshot()
	})

	it("renders the cartList array in no particular order", () => {
		const mockProducts = [
			{ id: 1, name: "Product 1" },
			{ id: 2, name: "Product 2" },
			{ id: 3, name: "Product 3" },
		]

		const cartList = { cartList: mockProducts }

		render(
			<ShopContext.Provider value={cartList}>
				<BrowserRouter>
					<CartList />
				</BrowserRouter>
			</ShopContext.Provider>
		)

		const allIDs = screen.getAllByTestId("product-id")

		expect(allIDs.length).toBe(mockProducts.length)
		expect(allIDs[0].textContent).toMatch(`${mockProducts[0].id}`)
		expect(allIDs[1].textContent).toMatch(`${mockProducts[1].id}`)
		expect(allIDs[2].textContent).toMatch(`${mockProducts[2].id}`)

		const allProductNames = screen.queryAllByTestId("product-name")

		expect(allProductNames.length).toBe(mockProducts.length)
	})
})

// Qué es lo más importante de este componente?
/*
- Renderizar lista de productos
- Si isEditOpen = true, que se muestre el componente EditCartItem
- Se renderiza "Checkout"

*/
