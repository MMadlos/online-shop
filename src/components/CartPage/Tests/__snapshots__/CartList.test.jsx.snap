// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CartList > renders the component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cart-list-container"
      >
        <div
          class="cart-product"
          data-product-id="1"
        >
          <a
            href="/product/1"
          >
            <img
              alt=""
              class="cart-product-image"
              src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
            />
          </a>
          <div
            class="cart-info-container"
          >
            <div
              class="cart-header-container"
            >
              <a
                href="/product/1"
              >
                <p
                  class="name"
                >
                  Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
                </p>
              </a>
              <i
                class="fa-regular fa-trash-can"
              />
            </div>
            <div
              class="product-quantity-container"
            >
              <p>
                Quantity: 0
              </p>
              <i
                class="fa-solid fa-pencil"
              />
            </div>
            <p
              class="price"
            >
              $ 
              109.95
            </p>
          </div>
        </div>
        <div
          class="divider"
        />
        <div
          class="cart-product"
          data-product-id="2"
        >
          <a
            href="/product/2"
          >
            <img
              alt=""
              class="cart-product-image"
              src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
            />
          </a>
          <div
            class="cart-info-container"
          >
            <div
              class="cart-header-container"
            >
              <a
                href="/product/2"
              >
                <p
                  class="name"
                >
                  Mens Casual Premium Slim Fit T-Shirts 
                </p>
              </a>
              <i
                class="fa-regular fa-trash-can"
              />
            </div>
            <div
              class="product-quantity-container"
            >
              <p>
                Quantity: 0
              </p>
              <i
                class="fa-solid fa-pencil"
              />
            </div>
            <p
              class="price"
            >
              $ 
              22.3
            </p>
          </div>
        </div>
      </div>
      <div
        class="checkout-container"
      >
        <img
          alt=""
          class="shipping-image"
          src="/src/assets/free-shipping-tiny.png"
        />
        <div
          class="cost-list"
        >
          <div
            class="subtotal"
          >
            <p>
              Number of items
            </p>
            <p>
              0
            </p>
          </div>
          <div
            class="subtotal"
          >
            <p>
              Subtotal
            </p>
            <p>
              $ 
              0.00
            </p>
          </div>
          <div
            class="divider"
          />
          <div
            class="shipping-cost"
          >
            <p>
              Shipping cost
            </p>
            <div
              class="chip accent"
            >
              <p>
                FREE
              </p>
            </div>
          </div>
          <div
            class="total-amount"
          >
            <p>
              TOTAL
            </p>
            <p>
              $ 
              0.00
            </p>
          </div>
          <p
            class="taxes"
          >
            (Taxes included)
          </p>
        </div>
        <button
          id="checkout"
        >
          Checkout
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cart-list-container"
    >
      <div
        class="cart-product"
        data-product-id="1"
      >
        <a
          href="/product/1"
        >
          <img
            alt=""
            class="cart-product-image"
            src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
          />
        </a>
        <div
          class="cart-info-container"
        >
          <div
            class="cart-header-container"
          >
            <a
              href="/product/1"
            >
              <p
                class="name"
              >
                Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops
              </p>
            </a>
            <i
              class="fa-regular fa-trash-can"
            />
          </div>
          <div
            class="product-quantity-container"
          >
            <p>
              Quantity: 0
            </p>
            <i
              class="fa-solid fa-pencil"
            />
          </div>
          <p
            class="price"
          >
            $ 
            109.95
          </p>
        </div>
      </div>
      <div
        class="divider"
      />
      <div
        class="cart-product"
        data-product-id="2"
      >
        <a
          href="/product/2"
        >
          <img
            alt=""
            class="cart-product-image"
            src="https://fakestoreapi.com/img/71-3HjGNDUL._AC_SY879._SX._UX._SY._UY_.jpg"
          />
        </a>
        <div
          class="cart-info-container"
        >
          <div
            class="cart-header-container"
          >
            <a
              href="/product/2"
            >
              <p
                class="name"
              >
                Mens Casual Premium Slim Fit T-Shirts 
              </p>
            </a>
            <i
              class="fa-regular fa-trash-can"
            />
          </div>
          <div
            class="product-quantity-container"
          >
            <p>
              Quantity: 0
            </p>
            <i
              class="fa-solid fa-pencil"
            />
          </div>
          <p
            class="price"
          >
            $ 
            22.3
          </p>
        </div>
      </div>
    </div>
    <div
      class="checkout-container"
    >
      <img
        alt=""
        class="shipping-image"
        src="/src/assets/free-shipping-tiny.png"
      />
      <div
        class="cost-list"
      >
        <div
          class="subtotal"
        >
          <p>
            Number of items
          </p>
          <p>
            0
          </p>
        </div>
        <div
          class="subtotal"
        >
          <p>
            Subtotal
          </p>
          <p>
            $ 
            0.00
          </p>
        </div>
        <div
          class="divider"
        />
        <div
          class="shipping-cost"
        >
          <p>
            Shipping cost
          </p>
          <div
            class="chip accent"
          >
            <p>
              FREE
            </p>
          </div>
        </div>
        <div
          class="total-amount"
        >
          <p>
            TOTAL
          </p>
          <p>
            $ 
            0.00
          </p>
        </div>
        <p
          class="taxes"
        >
          (Taxes included)
        </p>
      </div>
      <button
        id="checkout"
      >
        Checkout
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
